### ISSUE TITLE
SkateCat homebrew game refuses to boot properly after playing back a certain movie

### ISSUE THREAD

**Issue by despoa** [TASEmulators/fceux#616]
On FCEUX 2.6.5, the homebrew game [SkateCat](https://dale-coop.itch.io/skatecat-sj-games-nes) will no longer boot properly after playing back this [specific movie](https://tasvideos.org/7985S). Until the .sav file for the game is deleted, the game will boot to a grey screen. This doesn't not happen in 2.6.4 and 2.2.3.

edit: I can confirm that the exact same thing happens with the homebrew game [Rescue Time](https://dale-coop.itch.io/rescue-time-by-sj-games-nes) using [this movie](https://tasvideos.org/7984S). 2.6.5 doesn't work, 2.6.4 and 2.2.3 work.

**Comment by thor2016** [comment]
I traced the issue to a change that was made to the unrom512 mapper. Code was added to save the flash memory using the save RAM mechanism. However, when movie record/play is active, the following code:
```
// if we are in a movie, then reset the saveram
extern int disableBatteryLoading;
if (disableBatteryLoading)
	GameInterface(GI_RESETSAVE);
```
will eventually call `FCEU_ClearGameSave` which zero the save RAM (clearing the flash to zero in the process). I've commented out the hooks to the flash save RAM logic (commit 20b98285) until this issue can be fixed. The flash save RAM code was added by @ClusterM . Maybe he has some ideas on how to fix it. Anyway, with this flash saving mechanism disabled, the movie you provided played as expected.
